src/fpm/toml.f90

```diff
@@ -207,30 +228,64 @@ subroutine load_from_file(self, file, error)
         if (.not. exist) return
 
         open (file=file, newunit=unit)
-        call self%load(unit, error)
+        call self%load(unit, error, json)
         close (unit)
     end subroutine load_from_file
 
     !> Read dependency tree from file
-    subroutine load_from_unit(self, unit, error)
+    subroutine load_from_unit(self, unit, error, json)
         !> Instance of the dependency tree
         class(serializable_t), intent(inout) :: self
         !> File name
         integer, intent(in) :: unit
         !> Error handling
         type(error_t), allocatable, intent(out) :: error
+        !> Optional JSON format
+        logical, optional, intent(in) :: json
 
-        type(toml_error), allocatable :: parse_error
+        type(toml_error), allocatable :: toml_error
         type(toml_table), allocatable :: table
+        type(json_lexer) :: lexer
+        logical :: is_json
 
-        call toml_load(table, unit, error=parse_error)
+        is_json = .false.; if (present(json)) is_json = json
 
-        if (allocated(parse_error)) then
-          allocate (error)
-          call move_alloc(parse_error%message, error%message)
-          return
-        end if
+        if (is_json) then
+
+           !> Deactivate JSON deserialization for now
+           call fatal_error(error, 'JSON deserialization option is not yet available')
+           return
+
+           !> init JSON interpreter
+           call new_lexer_from_unit(lexer, unit, toml_error)
+           if (allocated(toml_error)) then
+              allocate (error)
+              call move_alloc(toml_error%message, error%message)
+              return
+           end if
+
+           !> Parse JSON to TOML table
+           call parse(lexer, table, error=toml_error)
+           if (allocated(toml_error)) then
+              allocate (error)
+              call move_alloc(toml_error%message, error%message)
+              return
+           end if
+
+        else
+
+           !> use default TOML parser
+           call toml_load(table, unit, error=toml_error)
+
+           if (allocated(toml_error)) then
+              allocate (error)
+              call move_alloc(toml_error%message, error%message)
+              return
+           end if
+
+        endif
```

Comment:
@awvwgk I'm trying to implement a JSON interface on top of the TOML one, using Jonquil, can I ask you if this approach is the right one? the latest release solves some comma issues, but the loaded TOML table still can't find any keys, I must be doing something wrong.